File: cgi-bin/savemap.py
#!/usr/bin/python

import sys
import cgi
import os

print "Content-Type: text/html; charset=UTF-8\n"

form = cgi.FieldStorage()
if "name" not in form:
    print "Please specify map name"
    sys.exit()

if "data" not in form:
    print "Missing map data"
    sys.exit()

root = os.getcwd()
fileName = root + "/data/maps/" + form.getlist("name")[0]
content = form.getlist("data")[0]

file = open(fileName, 'w')
file.write(content)
file.close()

File: cgi-bin/savemap.php
<?php
header('Content-Type: application/json');

if (!isset($_POST['name'])) {
    echo "Specify the name parameter (file name)";
    exit;
}

if (!isset($_POST['data'])) {
    echo "Specify the data parameter (map data)";
    exit;
}

$path = dirname(__DIR__) . DIRECTORY_SEPARATOR . "data" . DIRECTORY_SEPARATOR;
$file = $path . "maps" . DIRECTORY_SEPARATOR . $_POST['name'];

$data = $_POST['data'];
file_put_contents($file, $data);
chmod($file, 0666);
